examples/vision/xray_classification_with_tpus.py

```diff
@@ -0,0 +1,489 @@
+"""
+Title: Pneumonia Classification with TPUs
+Author: Amy MiHyun Jang
+Date created: 2020/07/28
+Last modified: 2020/07/28
+Description: Medical image classification using TPUs, tf.distribute, and keras.callbacks
+"""
+"""
+# 1. Introduction + Set-up
+
+Machine learning has a phenomenal range of applications, including in health and
+diagnostics. This tutorial will explain the complete pipeline from loading data to
+predicting results, and it will explain how to build an X-ray image classification model
+from scratch to predict whether an X-ray scan shows presence of pneumonia. This is
+especially useful during these current times as COVID-19 is known to cause pneumonia.
+
+This tutorial will explain how to utilize TPUs efficiently, load in image data, build and
+train a convolution neural network, finetune and regularize the model, and predict
+results. Data augmentation is not included in the model because X-ray scans are only
+taken in a specific orientation, and variations such as flips and rotations will not
+exist in real X-ray images. For a tutorial on image data augmentation, check out this
+[tutorial](https://www.kaggle.com/amyjang/tensorflow-data-augmentation-efficientnet).
+
+Run the following cell to load the necessary packages. Make sure to change the
+Accelerator to `TPU`.
+"""
+
+import re
+import os
+import numpy as np
+import pandas as pd
+import tensorflow as tf
+import matplotlib.pyplot as plt
+from sklearn.model_selection import train_test_split
+
+try:
+    tpu = tf.distribute.cluster_resolver.TPUClusterResolver()
+    print("Device:", tpu.master())
+    tf.config.experimental_connect_to_cluster(tpu)
+    tf.tpu.experimental.initialize_tpu_system(tpu)
+    strategy = tf.distribute.experimental.TPUStrategy(tpu)
+except:
+    strategy = tf.distribute.get_strategy()
+print("Number of replicas:", strategy.num_replicas_in_sync)
+
+print(tf.__version__)
+
+
+"""
+We need a Google Cloud link to our data to load the data using a TPU. While we're at it,
+we instantiate constant variables. It is generally better practice to use constant
+variables instead of hard-coding numbers.
+"""
+
+AUTOTUNE = tf.data.experimental.AUTOTUNE
+GCS_PATH = "gs://kds-63fdbf6403ae1d5e971e6cdbd4104b02e914cf6e90a178bd0e3990a8"
+BATCH_SIZE = 16 * strategy.num_replicas_in_sync
+IMAGE_SIZE = [180, 180]
+EPOCHS = 25
+
+"""
+# 2. Load the data
```

Comment:
For data loading, you can use:

```python
image_size = (180, 180)
batch_size = 32

train_ds = tf.keras.preprocessing.image_dataset_from_directory(
    "train",
    image_size=image_size,
    batch_size=batch_size,
)
val_ds = tf.keras.preprocessing.image_dataset_from_directory(
    "val",
    image_size=image_size,
    batch_size=batch_size,
)
test_ds = tf.keras.preprocessing.image_dataset_from_directory(
    "test",
    image_size=image_size,
    batch_size=batch_size,
)
class_names = train_ds.class_names
```

Reply:
I've tried using `image_dataset_from_directory` but it seems that it doesn't work with TPUs

Reply:
![image](https://user-images.githubusercontent.com/43710235/89088612-3945b180-d34e-11ea-95cb-1964409d321c.png)

Reply:
Error message:
```
InvalidArgumentError: NodeDef expected inputs 'string' do not match 0 inputs specified; Op<name=TensorSliceDataset; signature=components: -> handle:variant; attr=Toutput_types:list(type),min=1; attr=output_shapes:list(shape),min=1; is_stateful=true>; NodeDef: {{node TensorSliceDataset}} [Op:MapDataset]
```
Only the accelerator was changed. No change in code.

Reply:
This is not expected. Please file an issue so we can track this.